riva/proto/riva_nmt.proto

```diff
@@ -21,12 +24,88 @@ service RivaTranslation {
 
   // Translate text to text, from a source to a target language.  Currently source and target language fields is required, along with the model name.
   // Multiple texts may be passed per request up to the given batch size for the model, which is set at translation pipeline creation time.
-rpc TranslateText(TranslateTextRequest) returns (TranslateTextResponse) {}
+  rpc TranslateText(TranslateTextRequest) returns (TranslateTextResponse) {}
 
   // Lists the available language pairs and models names to be used for TranslateText
   rpc ListSupportedLanguagePairs(AvailableLanguageRequest) returns (AvailableLanguageResponse) {}
+
+  //streaming speech to text translation api.
+  rpc StreamingTranslateSpeechToText(stream StreamingTranslateSpeechToTextRequest)
+        returns (stream StreamingTranslateSpeechToTextResponse) {}
+
+  //offline speech to text translation api.
+  //rpc TranslateSpeechToText(TranslateSpeechToTextRequest) returns (TranslateSpeechToTextResponse) {}
+
+  //Returns the parameters for the configurated pipelines.
+  //rpc GetTranslateSpeechToTextConfig(GetTranslateSpeechToTextConfigRequest) returns (GetTranslateSpeechToTextConfigResponse) {}
+
+  rpc StreamingTranslateSpeechToSpeech(stream StreamingTranslateSpeechToSpeechRequest)
+    returns (stream StreamingTranslateSpeechToSpeechResponse) {}
+
+  //offline speech to text translation api.
+  //rpc TranslateSpeechToSpeech(TranslateSpeechToSpeechRequest) returns (TranslateSpeechToSpeechResponse) {}
+
+
 }
 
+/*
+* Configuration for Translate S2S.  reuse existing protos from other services.
+*/
+message StreamingTranslateSpeechToSpeechConfig {
+   nvidia.riva.asr.StreamingRecognitionConfig asrConfig = 1;  //from riva_asr.proto
+   SynthesizeSpeechConfig ttsConfig = 2; //from riva_tts.proto
+   string source_language_code = 3;
+   string target_language_code = 4;
+   AudioEncoding t = 5;
```

Comment:
AudioEncoding is part of both asr config and tts config.  Why have a separate parameter here?  I guess I could ask the same question for source_language_code.

Reply:
source/target lang are the NMT parameters.  They both could be infered from their respective ASR and TTS configs.
AudioEncoding will be removed.

Reply:
added nmt config message, removed audioencoding.